helics/capi.py

```diff
@@ -717,9 +718,33 @@ class HelicsFederateState(IntEnum):
 helics_state_pending_finalize = HelicsFederateState.PENDING_FINALIZE
 
 
+def generate_cleanup_callback(obj):
+
+    if issubclass(obj.__class__, HelicsFederate):
+        f = loadSym("helicsFederateFree")
+    elif issubclass(obj.__class__, HelicsFederateInfo):
+        f = loadSym("helicsFederateInfoFree")
+    elif issubclass(obj.__class__, HelicsBroker):
+        f = loadSym("helicsBrokerFree")
+    elif issubclass(obj.__class__, HelicsCore):
+        f = loadSym("helicsCoreFree")
+    elif issubclass(obj.__class__, HelicsQuery):
+        f = loadSym("helicsQueryFree")
+    # elif issubclass(obj.__class__, HelicsMessage):
+    #     f = loadSym("helicsMessageFree")
+    else:
```

Comment:
Possible silent failure? Perhaps worth a log message?

Reply:
Resolved in https://github.com/GMLC-TDC/pyhelics/pull/26/commits/c226212d2d6cac119f85d477ce360e626e8ccbd3.

Reply:
Sorry I wasn't clear. I was referring to the final `else` case that effectively does nothing. If a class in the future is created and the developer neglects to add an `elif` case here, the final `else` will run silently, which means some cleanup could be accidentally neglected.

Reply:
Ah. I understand what you meant now. There are a few classes that are handles to helics objects but don’t have free functions because they are cleaned up when the federate is destroyed. I can add explicit checks for all of those.

Reply:
Resolved in https://github.com/GMLC-TDC/pyhelics/pull/26/commits/b2998fbf7149d0229161375d8454fc825fd5cefd.